### PR TITLE
Visitor Opaque Authentication

### PR DESCRIPTION
**Pull request by igorkravchenko**
Add visitor opaque authentication. This is accompanying addition on top of Core SDK, aimed to bring VOA (also known as *Direct ID Token Authentication*) to Widgets:

- Currently authentication is not allowed during ongoing engagement (this restriction may be lifted in future).
- In order to authenticate or deauthenticate SDK configuration must be accomplished.

MOB-1492

### CODE REVIEW

File: GliaWidgetsTests/ViewModel/Chat/Storage/AuthenticatedChatStorageTests.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    func test_isNewMessageReturnsTrueForNotStoredMessage() {
+        let messageId = "messageId"
+        XCTAssertTrue(storage.isNewMessage(.mock(id: messageId)))
+    }
+
+    func test_isNewMessageReturnsFalseForStoredMessage() {
+        let messageId = "messageId"
+        let queueId = "queueID"
```
Comment (gersonnoboa) [inline GliaWidgetsTests/ViewModel/Chat/Storage/AuthenticatedChatStorageTests.swift:63]: maybe messageId and queueId can be set as lets at the top? seems like they are the same for all

File: Podfile [hunk 2]
```diff
@@ -23,4 +23,4 @@ end
 
 target 'SnapshotTests' do
   pod 'AccessibilitySnapshot', '0.5.0'
-end
\ No newline at end of file
+end
```
Comment (gersonnoboa) [inline Podfile:26]: Nice

**Review by gersonnoboa — APPROVED** [review]

**Comment by gersonnoboa** [comment]
However, a description in the commit would be useful.

File: Podfile (hunk 2 above)
Comment (igorkravchenko) [reply Podfile:26]: That is a part of 71208e1c8e0961100c29d77eb65e78dbabe24100 where I remove linked locally Core SDK source via Cocoapods (committed that by mistake).

**Review by dukhovnyi — APPROVED** [review]

**Comment by dukhovnyi** [comment]
!squash

**Pull request merged**